File: swap_backend.js
const { WalletClient } = require('@arqma/arqma-rpc');

class XEQSwapBackend {
    constructor() {
        // Source wallet client (old XEQ)
        this.sourceWalletClient = new WalletClient({
            host: '127.0.0.1',
            port: 18090,  // Source wallet RPC port
            protocol: 'http'
        });

        // Target wallet client (new XEQ)
        this.targetWalletClient = new WalletClient({
            host: '127.0.0.1',
            port: 18096,  // Target wallet RPC port
            protocol: 'http'
        });
    }

    async getSourceBalance() {
        try {
            const result = await this.sourceWalletClient.getBalance();
            return result;
        } catch (error) {
            console.error('Error getting source balance:', error);
            return null;
        }
    }

    async getTargetBalance() {
        try {
            const result = await this.targetWalletClient.getBalance();
            return result;
        } catch (error) {
            console.error('Error getting target balance:', error);
            return null;
        }
    }

    async transferXEQ(amount, fromSource = true) {
        try {
            if (fromSource) {
                // Transfer from source to target
                const targetAddress = await this.targetWalletClient.getAddress();
                const result = await this.sourceWalletClient.transfer({
                    destinations: [{
                        address: targetAddress,
                        amount: amount
                    }]
                });
                return result;
            } else {
                // Transfer from target to source
                const sourceAddress = await this.sourceWalletClient.getAddress();
                const result = await this.targetWalletClient.transfer({
                    destinations: [{
                        address: sourceAddress,
                        amount: amount
                    }]
                });
                return result;
            }
        } catch (error) {
            console.error('Error transferring XEQ:', error);
            return null;
        }
    }
}

// Example usage
async function main() {
    const swapBackend = new XEQSwapBackend();
    
    try {
        const sourceBalance = await swapBackend.getSourceBalance();
        const targetBalance = await swapBackend.getTargetBalance();
        
        console.log('Source wallet balance:', sourceBalance);
        console.log('Target wallet balance:', targetBalance);
    } catch (error) {
        console.error('Error in main:', error);
    }
}

if (require.main === module) {
    main();
}

module.exports = XEQSwapBackend; 

File: swap_backend.py
import sys
import os
from framework.wallet import Wallet

class XEQSwapBackend:
    def __init__(self):
        # Source wallet client (old XEQ)
        self.source_wallet = Wallet(
            protocol='http',
            host='127.0.0.1',
            port=18090  # Source wallet RPC port
        )
        
        # Target wallet client (new XEQ)
        self.target_wallet = Wallet(
            protocol='http',
            host='127.0.0.1',
            port=18096  # Target wallet RPC port
        )

    def get_source_balance(self):
        """Get balance from source wallet"""
        try:
            result = self.source_wallet.get_balance()
            return result
        except Exception as e:
            print(f"Error getting source balance: {e}")
            return None

    def get_target_balance(self):
        """Get balance from target wallet"""
        try:
            result = self.target_wallet.get_balance()
            return result
        except Exception as e:
            print(f"Error getting target balance: {e}")
            return None

    def transfer_xeq(self, amount, from_source=True):
        """Transfer XEQ between wallets"""
        try:
            if from_source:
                # Transfer from source to target
                destinations = [{
                    'address': self.target_wallet.get_address(),
                    'amount': amount
                }]
                result = self.source_wallet.transfer(destinations)
            else:
                # Transfer from target to source
                destinations = [{
                    'address': self.source_wallet.get_address(),
                    'amount': amount
                }]
                result = self.target_wallet.transfer(destinations)
            return result
        except Exception as e:
            print(f"Error transferring XEQ: {e}")
            return None

def main():
    swap_backend = XEQSwapBackend()
    
    # Example usage
    source_balance = swap_backend.get_source_balance()
    target_balance = swap_backend.get_target_balance()
    
    print(f"Source wallet balance: {source_balance}")
    print(f"Target wallet balance: {target_balance}")

if __name__ == "__main__":
    main() 